devbuild.sh

```diff
@@ -1,135 +0,0 @@
-#!/bin/bash
```

Comment:
@christopherwharrop-noaa The community develop has decided that they would like to support a `devbuild.sh`, whether that's a good idea or not, I probably shouldn't comment on.

I'm curious if we could potentially adopt and adapt [that one](https://github.com/ufs-community/ufs-srweather-app/blob/develop/devbuild.sh) for the test here?

Reply:
I am ok with utility "helper" scripts that make tedious tasks easier as long as they don't over-engineer or cause portability problems.  If the community wants to use one, then RRFS needs to use whatever the community chooses.  If customizations are required for RRFS, we'll have to feed those back to the community when it gets merged there.

Reply:
As this PR stands now, there really isn't any need for a `devbulid.sh` script because the build is so trivial, requiring only the basic "Getting Started" instructions.  Right now, two short commands from "Getting Started" do what that entire `devbuild.sh` script was doing.

Reply:
Okay! :)

Reply:
I have added the `devbuild.sh` utility from the current ufs-community version of the App.  I tested it out and tried several options.  It's a pretty nice utility for the most part.  The one thing I do not like is its auto detection of the platform.  The very small value added by the platform detection is far outweighed by the non-portability, maintenance cost, and security concerns associated with maintaining a script full of hardcoded fully qualified hostnames that are then published on a public repository.  All that work and headache to avoid typing `--platform=hera` is just not worth it.  However, I have pushed what the community is currently using despite those objections.

Reply:
FYI: My typical usage for this script would be: `./devbuild.sh --platform=hera --compiler=intel --verbose --build-jobs=8 --clean`.  Substitute `hera` for the name of the platform you are using, of course.